mod/consensus-types/pkg/types/payload.go

```diff
@@ -273,7 +273,7 @@ func (p *ExecutionPayload) GetTree() (*fastssz.Node, error) {
 /* -------------------------------------------------------------------------- */
 
 // MarshalJSON marshals as JSON.
-func (p *ExecutionPayload) MarshalJSON() ([]byte, error) {
```

Comment:
q: not a fan of mixing value and pointer receivers. Should we consider having the whole struct using value receivers (despite all the coping around)?

Reply:
I would bias towards always using pointers. Copying full blocks gets expensive.